local_setup/sha_generator.go

```diff
@@ -0,0 +1,254 @@
+	Images        *Image `json:"Images"`
+}
+
+var (
+	VERBOSE  = false
+	DBCONFIG *DatabaseConfig
+)
+
+func generateSHA256(count int) (sha []string) {
+	sha = make([]string, count)
+	secret := []byte("mysecret")
+	for i := 0; i < count; i++ {
+		// Create a new HMAC by defining the hash type and the key (as byte array)
+		h := hmac.New(sha256.New, secret)
+		// Write Data to it
+		h.Write([]byte(time.Now().String()))
+		// Get result and encode as hexadecimal string
+		sha[i] = "sha256:" + hex.EncodeToString(h.Sum(nil))
+	}
+	return
+}
+
+func produce(shas []string, account, org int, cluster, broker, topic string) {
+	producer, err := sarama.NewSyncProducer([]string{broker}, nil)
+	if err != nil {
+		fmt.Printf("couldn't connect to Kafka broker %v\n", broker)
+		os.Exit(ExitStatusKafkaBrokerError)
+	}
+
+	kafkaMsg := KafkaMessage{
+		AccountNumber: account,
+		Organization:  org,
+		ClusterName:   cluster,
+		Images: &Image{
+			ImageCount: len(shas),
+		},
+	}
+
+	images := make(JSONContent, len(shas))
+
+	//empty content, as long as it is a valid JSON object
+	content := json.RawMessage("{}")
+	for _, msg := range shas {
+		images[msg] = &content
+	}
+
+	kafkaMsg.Images.Digests = &images
+
+	if VERBOSE {
+		fmt.Printf("content of Kafka message to produce:\n\torg: %v\n\taccount: %v\n\tcluster:%v\n\tDigests:%v\n", kafkaMsg.Organization, kafkaMsg.AccountNumber, kafkaMsg.ClusterName, kafkaMsg.Images.Digests)
+	}
+
+	jsonBytes, err := json.Marshal(kafkaMsg)
+	if err != nil {
+		fmt.Println("couldn't turn Kafka message into valid JSON")
+		fmt.Printf("error: %v\n", err)
+		fmt.Printf("kafka message: %v\n", kafkaMsg)
+		os.Exit(ExitStatusKafkaProducerError)
+	}
+
+	producerMsg := &sarama.ProducerMessage{
+		Topic: topic,
+		Value: sarama.ByteEncoder(jsonBytes),
+	}
+
+	partitionID, offset, err := producer.SendMessage(producerMsg)
+	if err != nil {
+		fmt.Println(err.Error())
+		fmt.Println("failed to produce message to Kafka")
+	} else {
+		if VERBOSE {
+			fmt.Printf("message sent to partition %d at offset %d\n", partitionID, offset)
+		}
+	}
+}
+
+// Setup DB gets the environment variables ONLY from config.yml
+func setupDB() (dataSource string) {
+	viper.AddConfigPath(".")
+	viper.SetConfigName("config")
+	viper.SetConfigType("yml")
+
+	if err := viper.ReadInConfig(); err != nil {
+		fmt.Println(err.Error())
+		fmt.Println("Error reading config file")
+		os.Exit(ExitStatusDBConfigError)
+	}
+
+	if err := viper.Unmarshal(&DBCONFIG); err != nil {
+		fmt.Println(err.Error())
+		fmt.Println("Unable to decode into config struct")
+		os.Exit(ExitStatusDBConfigError)
+	}
+
+	dataSource = fmt.Sprintf(
+		"postgresql://%v:%v@%v:%v/%v?%v",
+		DBCONFIG.PGUsername,
+		DBCONFIG.PGPassword,
+		DBCONFIG.PGHost,
+		DBCONFIG.PGPort,
+		DBCONFIG.PGDBName,
+		DBCONFIG.PGParams,
+	)
+	return
+}
+func prepareInsertDigestsStatement(shas []string) (statement string, statementArgs []interface{}) {
+	statement = `INSERT INTO image (digest, health_index) VALUES %s`
```

Comment:
The health index column was removed from DB for now.

Reply:
I thought I updated that. I probably messed it up in between cherry picks